### DISCUSSION TITLE
How to Visualize the Model Graph of a PyG model in Tensorboard

### DISCUSSION THREAD

**Discussion by AnimeshSinha1309** [pyg-team/pytorch_geometric#3087] [Q&A] [answered]
I have used both PyTorch and Tensorboard, and logging the model graph for any Graph Neural Network is being a problem everywhere, because the graphing function takes a dummy input, and that input must be a tensor, not a PyG Graph.
Any suggestions on how to subvert this and see my model graphs on Tensorboard.

**Comment by rusty1s** [comment, marked as answer]
Tensorboard should work out-of-the-box, but you should input tensors to your model rather than the `data` object. See [here](https://github.com/rusty1s/pytorch_geometric/blob/master/examples/tensorboard_logging.py) for an example.

**Comment by AnimeshSinha1309** [comment]
Thanks a lot, the mistake I was making was that I was passing in the data object into the model whereas I had to pass in tensors, after extracting from the data.
